test/api/clustering_methods_test.cpp

```diff
@@ -0,0 +1,336 @@
+                    Breakend{chrom2, chrom2_position1 - 1, strand::reverse}, ""_dna5, read_name_4}, //cluster 2
+        Junction{Breakend{chrom1, chrom1_position1 + 92, strand::forward},
+                    Breakend{chrom2, chrom2_position1 + 3, strand::forward}, ""_dna5, read_name_5}, //cluster 3
+        Junction{Breakend{chrom1, chrom1_position2 - 2, strand::forward},
+                    Breakend{chrom2, chrom1_position3 + 8, strand::reverse}, ""_dna5, read_name_6}, //cluster 4
+        Junction{Breakend{chrom1, chrom1_position2 + 3, strand::forward},
+                    Breakend{chrom2, chrom1_position3 - 1, strand::reverse}, ""_dna5, read_name_7}, //cluster 4
+        Junction{Breakend{chrom1, chrom1_position2 + 6, strand::forward},
+                    Breakend{chrom2, chrom1_position3 + 2, strand::reverse}, ""_dna5, read_name_8} //cluster 4
+    };
+
+    std::sort(input_junctions.begin(), input_junctions.end());
+    return input_junctions;
+}
+
+
+TEST(hierarchical_clustering, partitioning)
+{
+    std::vector<Junction> input_junctions = prepare_input_junctions();
+    std::vector<std::vector<Junction>> partitions = partition_junctions(input_junctions);
+
+    std::vector<std::vector<Junction>> expected_partitions
+    {   
+        {   Junction{Breakend{chrom1, chrom1_position1 - 5, strand::forward},
+                     Breakend{chrom2, chrom2_position1 + 8, strand::forward}, ""_dna5, read_name_1}, //cluster 1
+            Junction{Breakend{chrom1, chrom1_position1 + 2, strand::forward},
+                     Breakend{chrom2, chrom2_position1 - 3, strand::forward}, ""_dna5, read_name_2}, //cluster 1
+            Junction{Breakend{chrom1, chrom1_position1 + 9, strand::forward},
+                     Breakend{chrom2, chrom2_position1 + 1, strand::forward}, ""_dna5, read_name_3}, //cluster 1
+        },
+        {   Junction{Breakend{chrom1, chrom1_position1 + 5, strand::forward},
+                     Breakend{chrom2, chrom2_position1 - 1, strand::reverse}, ""_dna5, read_name_4}, //cluster 2
+        },
+        {   Junction{Breakend{chrom1, chrom1_position1 + 92, strand::forward},
+                     Breakend{chrom2, chrom2_position1 + 3, strand::forward}, ""_dna5, read_name_5}, //cluster 3
+        },
+        {   Junction{Breakend{chrom1, chrom1_position2 - 2, strand::forward},
+                     Breakend{chrom2, chrom1_position3 + 8, strand::reverse}, ""_dna5, read_name_6}, //cluster 4
+            Junction{Breakend{chrom1, chrom1_position2 + 3, strand::forward},
+                     Breakend{chrom2, chrom1_position3 - 1, strand::reverse}, ""_dna5, read_name_7}, //cluster 4
+            Junction{Breakend{chrom1, chrom1_position2 + 6, strand::forward},
+                     Breakend{chrom2, chrom1_position3 + 2, strand::reverse}, ""_dna5, read_name_8} //cluster 4
+        }
+    };
+
+    EXPECT_EQ(expected_partitions.size(), partitions.size());
+
+    if (expected_partitions.size() == partitions.size())
+    {
```

Comment:
Use `ASSERT_` instead of `EXPECT_EQ` to force gtest to abort this test. This way you can skip your if.
```suggestion
    ASSERT_EQ(expected_partitions.size(), partitions.size());
```

Reply:
Great suggestion! 👍